src/drt/src/io/io_parser_helper.cpp

```diff
@@ -551,8 +552,186 @@ void io::Parser::convertLef58MinCutConstraints()
   }
 }
 
+inline void getTrackLocs(bool isHorzTracks,
```

Comment:
Is this a copy of FlexDRWorker::getTrackLocs ?  Please have one copy 
No need for inline in a .cpp

Reply:
not a copy

Reply:
Perhaps not a literal copy but it seems to do the same thing.  Is there some difference I'm missing?

Reply:
When I looked at FlexDRWorker::getTrackLocs, I didn't understand what it was trying to do. First there is this part:
https://github.com/The-OpenROAD-Project/OpenROAD/blob/6a66a6b07a3231b9400ed35d0f4e71eace5f5f4f/src/drt/src/dr/FlexDR_init.cpp#L1394-L1401
trackLocs is a set.  At the first line, you already add the trackLoc to the set. The if condition after that is completely useless. And this is the only part where the bool isHorzTracks is used, so it is completely useless as well. 
I didn't focus on this when I started writing the check, so I just changed the implementation, but when you mentioned it, I came up with the mentioned findings. However, I am not sure which part is incorrect, inserting the trackLoc regardless of its direction or validating on isHorzTracks bool.
I am gonna start a GH issue to look at this when I have time.

Reply:
Ok I'll approve here and you can figure this out when address your PR.  My goal is one method that does this correctly and is shared.